lib/Macros/Sources/SwiftMacros/SwiftifyImportMacro.swift

```diff
@@ -699,26 +702,28 @@ struct CountedOrSizedReturnPointerThunkBuilder: PointerBoundsThunkBuilder {
         "start"
       }
     var cast = try newType
+    var expr: ExprSyntax
     if nullable {
       if let optType = cast.as(OptionalTypeSyntax.self) {
         cast = optType.wrappedType
       }
-      return """
-      { () in
-        let _resultValue = \(call)
-        if unsafe _resultValue == nil {
-          return nil
-        } else {
-          return unsafe \(raw: cast)(\(raw: startLabel): _resultValue!, count: Int(\(countExpr)))
-        }
-      }()
-      """
+      expr =
+        """
+        { () in
+          let _resultValue = \(call)
+          if unsafe _resultValue == nil {
+            return nil
+          } else {
+            return unsafe _swiftifyOverrideLifetime(\(raw: cast)(\(raw: startLabel): _resultValue!, \(raw: countLabel): Int(\(countExpr))), copying: ())
+          }
+        }()
+        """
+    } else {
+      expr =
+        """
+        \(raw: cast)(\(raw: startLabel): \(call), \(raw: countLabel): Int(\(countExpr)))
+        """
     }
-    var expr = ExprSyntax(
-      """
-      \(raw: cast)(\(raw: startLabel): \(call), count: Int(\(countExpr)))
-      """
-    )
     if generateSpan {
       expr = "_swiftifyOverrideLifetime(\(expr), copying: ())"
```

Comment:
Won't think result in two calls to `_swiftifyOverrideLifetime()`, one inside the other? Is that necessary?

Reply:
Yes, it is necessary until we change the macro to not use immediately called closures for nullable pointers, because we need it each time a Span is returned from a function (which it technically is in the closure)